circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -299,6 +301,45 @@ void validate_inputs(DummyComposer& composer, PrivateInputs<NT> const& private_i
     NT::fr const start_public_call_stack_length = array_length(start.public_call_stack);
     NT::fr const start_new_l2_to_l1_msgs_length = array_length(start.new_l2_to_l1_msgs);
 
+    {
+        // Verify public key hash matches ethereum address of the sender
+        // TODO(Suyash): We want to perform this only for the base case, so move to "initial" PKC.
```

Comment:
Good point, it could be. In the current version, each of the private kernel is being passed on a `from` and `signing_key`, so I kept this check for each kernel iteration. But you're right, we could move it to just base case since that's what we will be doing eventually.

Reply:
Done.